src/umpire/ResourceManager.cpp

```diff
@@ -399,6 +405,40 @@ ResourceManager::move(void* ptr, Allocator allocator)
     return ptr;
   }
 
+#if defined(UMPIRE_ENABLE_NUMA)
+  {
+    // short circuit if allocator is a NUMA allocator
+    auto numa_alloc = std::dynamic_pointer_cast<strategy::NumaPolicy>(allocator.getAllocationStrategy());
```

Comment:
The `Allocator` might be wrapped in an `AllocationTracker`, so you will have to check twice